test/integration/testdata/custom_cluster_domain_config.yaml

```diff
@@ -0,0 +1,10 @@
+kind: Cluster
```

Comment:
I think this is a duplicate of what is in `test/integration/configs/cluster-domain.yaml`

Reply:
Yep, It is. So, I updated `release.yaml` to use `test/integration/configs/cluster-domain.yaml` but I don't see a similar test in `kind_integration.yaml`. So, are we running these custom domain tests only for release? :)

Reply:
Opened #4668 to fix this separately from this change!